spring-integration-core/src/main/java/org/springframework/integration/context/IntegrationObjectSupport.java

```diff
@@ -268,7 +268,7 @@ public ConversionService getConversionService() {
 		return this.conversionService;
 	}
 
-	protected void setConversionService(ConversionService conversionService) {
+	public void setConversionService(ConversionService conversionService) {
```

Comment:
Looks like you didn't run tests in the project after this change.
The `RouterConcurrencyTest` is going to fail.

Reply:
This is strange. Gradle output a message on the console but didn't record it as an error:
```
> Task :spring-integration-core:compileTestJava
/home/christopher/git/spring/spring-integration/spring-integration-core/src/test/java/org/springframework/integration/router/RouterConcurrencyTest.java:60: error: setConversionService(ConversionService) in <anonymous org.springframework.integration.router.RouterConcurrencyTest$1> cannot override setConversionService(ConversionService) in IntegrationObjectSupport
                        protected void setConversionService(ConversionService conversionService) {
                                       ^
  attempting to assign weaker access privileges; was public

> Task :spring-integration-kafka:compileKotlin
```

Reply:
This does bring up a concern, though, that this is a breaking change for implementers (SemVer minor). Not sure how this changes whether it's worth accepting.

Reply:
Oh, right! I wouldn't worry too much for version `6.0.1`, since it is unlikely this method is overridden without increasing access, but still bad.
I worry more about the version `5.5.x` where indeed a bytecode must not be made incompatible.
Therefore we can accept your change, but for consistency only in the next `6.1.0` version.
For now we are stuck with a workaround to extend the class to be able to increasing its access.
Although I believe with an XML configuration it should work even with a `protected` modifier 😄 

I suggest to close this PR since it is not appropriate for the code base it is requested.

Thank you for spotting this!

Reply:
This would be appreciated for the 6.x branch, but for now I can reflectively access it for our specific use case (we're using Groovy anyway).